jerry-core/ecma/base/ecma-globals.h

```diff
@@ -2112,6 +2121,7 @@ typedef struct
   ecma_object_t header; /**< header part */
   ecma_value_t target; /**< [[ProxyTarget]] internal slot */
   ecma_value_t handler; /**< [[ProxyHandler]] internal slot */
+  uint8_t flags; /**< Store ecma_proxy_flags_t bits */
```

Comment:
This increases the struct by 8 bytes. Could we use tags to store these flags?

Reply:
I've tested and the size only increased by 4 bytes. Still we could store the flags as tags, but on which value? on the target? on the handler? on the proxy? If I understand correctly the tags would require a `jmem_cpointer_t` typed value.

Reply:
Well maybe this current solution using the proplist_cp to store the tags could work.

Reply:
If target or handler is an object, you can use the 3 bottom bits for free. We have a macro system for handling these bits. See byte-code functions.

Reply:
We need to keep the information even if the proxy is revoked and at that point the target/handler is an `ECMA_VALUE_NULL`.